docs/sql-programming-guide.md

```diff
@@ -1894,6 +1894,8 @@ working with timestamps in `pandas_udf`s to get the best performance, see
 
   - In PySpark, when creating a `SparkSession` with `SparkSession.builder.getOrCreate()`, if there is an existing `SparkContext`, the builder was trying to update the `SparkConf` of the existing `SparkContext` with configurations specified to the builder, but the `SparkContext` is shared by all `SparkSession`s, so we should not update them. Since 3.0, the builder come to not update the configurations. This is the same behavior as Java/Scala API in 2.3 and above. If you want to update them, you need to update them prior to creating a `SparkSession`.
 
+  - In Spark version 2.4 and earlier, HAVING without GROUP BY is treated as WHERE. This means, `SELECT 1 FROM range(10) HAVING true` is executed as `SELECT 1 FROM range(10) WHERE true`  and returns 10 rows. This violates SQL standard, and has been fixed in Spark 3.0. Since Spark 3.0, HAVING without GROUP BY is treated as a global aggregate, which means `SELECT 1 FROM range(10) HAVING true` will return only one row.
```

Comment:
shall we backport it to 2.4?

Reply:
For such a correctness issue, I think we should merge it to the 2.4 release

Reply:
You will need to feature flag it if you port it to 2.4. People might rely on its current behavior.

Reply:
Yes. We should add a legacy SQLConf